.github/workflows/api-ci.yml

```
@@ -0,0 +1,74 @@
+name: API CI
+
+on:
+  pull_request:
+    branches:
+      - master
+    paths:
+      - 'driver/**'
+      - 'passenger/**'
+      - 'rating/**'
+      - 'ride/**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build with Maven
+        run: mvn -B clean install -DskipTests
+
+  test:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+          cache: maven
+      - name: Set filters
+        uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            driver:
+              - 'driver/**'
+            passenger:
+              - 'passenger/**'
+            rating:
+              - 'rating/**'
+            ride:
+              - 'ride/**'
+      - name: Run unit and integration tests DRIVER
+        if: steps.changes.outputs.driver == 'true'
+        run: |
+          cd driver 
+          mvn -B clean install
+          cd ..
```

Как дополнительный вариант:
```suggestion
          mvn -B clean install --file driver/pom.xml
```